src/sidebar/components/MentionSuggestionsPopover.tsx

```diff
@@ -42,8 +43,13 @@ function SuggestionItem({
       )}
       onClick={e => {
         e.stopPropagation();
-        onSelectUser(user);
+        onSelect();
       }}
+      // Using onMouseMove instead of onMouseEnter, so that if the mouse is
+      // already hovering this item, and then we change the highlighted
+      // suggestion via arrow keys, when the mouse is moved again, this item
+      // becomes highlighted without having to first leave it and enter again.
+      onMouseMove={onHighlight}
```

Comment:
This is going to lead to redundant calls to `onHighlight` and thus `onHighlightSuggestion`, but that might be OK in practice because `onHighlightSuggestion` is connected to a state setter which does nothing if the old and new value are the same.